msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionForumBean.java

```diff
@@ -615,7 +617,9 @@ public String getFormattedOpenDate(){
 	}
 
 	public String getHasRubric(){
-		return rubricsService.hasAssociatedRubric(RubricsConstants.RBCS_TOOL_GRADEBOOKNG, forum.getDefaultAssignName()) ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
+
+        Optional<AssociationTransferBean> rubricAssociation = rubricsService.getAssociationForToolAndItem(RubricsConstants.RBCS_TOOL_GRADEBOOKNG, forum.getDefaultAssignName(), forumManager.getContextForForumById(forum.getId()));
```

Comment:
@zhuoY121 simply switching to a method that is devoid of any security doesn't seem legitimate?

Reply:
@ern Yes, you bring up a valid point. This is also the problem I want to talk about. 

I was also initially uncertain about this aspect. However, considering the requirements stated in the Jira ticket, it appears that students should be able to view the presence of a rubric. n this specific case, where the goal is to determine if a rubric exists, checking permissions may not be necessary. 

I am sure that students should NOT be able to get the rubric, so the method just returns a boolean value and does not modify the rubric. I think ```hasAssociatedRubric``` should be used here, but it seems like ```hasAssociatedRubric``` does not allow students to know whether there is a rubric.